Sources/NIO/Codec.swift

```diff
@@ -33,6 +33,9 @@ public enum ByteToMessageDecoderError: Error {
     /// This error can be thrown by `ByteToMessageDecoder`s if there was unexpectedly some left-over data when the
     /// `ByteToMessageDecoder` was removed from the pipeline or the `Channel` was closed.
     case leftoverDataWhenDone(ByteBuffer)
+
+    /// This error can be thrown by `ByteToMessageDecoder`s if the incoming payload is larger than the max defined.
+    case payloadTooLarge
```

Comment:
this is major version breaking, you need to add a new error type for this

Reply:
no strong opinion, do you prefer a separate error to get in earlier?

Reply:
I think it'd be a good idea: if we don't use a separate error type this will have to sit here for probably at least a year.